x-pack/legacy/plugins/security/server/routes/api/v1/authenticate.js

```diff
@@ -82,6 +83,29 @@ export function initAuthenticateApi({ authc: { login, logout }, config }, server
     }
   });
 
+  server.route({
+    method: 'GET',
+    path: '/api/security/v1/oidc/implicit',
+    config: { auth: false },
+    async handler(request, h) {
+      const legacyConfig = server.config();
+      const basePath = legacyConfig.get('server.basePath');
+
+      const nonce = await generateCSPNonce();
+      const cspRulesHeader = createCSPRuleString(legacyConfig.get('csp.rules'), nonce);
+      return h.response(`
+        <script nonce="${nonce}">
+          const redirectURI = window.location.pathname + window.location.search + window.location.hash;
```

Comment:
__note:__ we can just pass entire `window.location.href` instead, it will also work and make this code a tad easier. Now when I look at this I can't think of any reason to not use entire `window.location.href` :see_no_evil: What do you think?

Reply:
That seems easy enough to be, and potentially less fragile.

Reply:
Yep, will switch to `window.location.href`.